xbmc/cores/VideoPlayer/DVDSubtitles/DVDSubtitleStream.cpp

```diff
@@ -95,16 +95,17 @@ bool CDVDSubtitleStream::Open(const std::string& strFile)
 }
 
 bool CDVDSubtitleStream::IsIncompatible(CDVDInputStream* pInputStream,
-                                        std::vector<char>& buf,
+                                        std::string& buf,
                                         size_t* bytesRead)
 {
   if (!pInputStream)
     return true;
 
   static const uint8_t vobsub[] = { 0x00, 0x00, 0x01, 0xBA };
 
-  int read =
-      pInputStream->Read(reinterpret_cast<uint8_t*>(buf.data()), static_cast<int>(buf.size()));
+  //! @todo: fix with c++17
+  int read = pInputStream->Read(const_cast<uint8_t*>(reinterpret_cast<const uint8_t*>(buf.data())),
+                                static_cast<int>(buf.size()));
```

Comment:
Like in my other comment, who guarantees that there will be a `'\0'` at the end? Thinking also about corrupt files here... `std::vector` looks like the correct way to me.

Reply:
I've updated it to use `std::vector`